### PR TITLE
[1.26.0] feat: CliTest

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

This PR introduces the `CliTest` base class which for usage of the CliRunner in tests. This class is present in open-autonomy tests as one of the base classes for tests, however can be used here to address various flaky cli tests, such as the following:
https://github.com/valory-xyz/open-aea/blob/bbd8fc00e1c5ffe5c1147c2d3e342a3ecf7645cf/tests/test_cli/test_run.py#L61
and 11 more such markers used in this single test_run.py file alone. 

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: aea/test_tools/click_testing.py [hunk 1]
```diff
@@ -133,3 +145,71 @@ def invoke(  # type: ignore
             exc_info=exc_info,  # type: ignore
             return_value=None,
         )
+
+
+class CliTest:
+    """Test `autonomy analyse abci` command."""
```
Comment (Karrenbelt) [inline aea/test_tools/click_testing.py:151]: aah docstring!

File: aea/test_tools/click_testing.py [hunk 2]
```diff
@@ -133,3 +145,71 @@ def invoke(  # type: ignore
...
+
+
+class CliTest:
+    """Test cli commands."""
+
+    t: Path
+    cwd: Path
+    cli_runner: CliRunner
+    cli_options: Tuple[str, ...]
+    cli: click.core.Group = aea_cli
```
Comment (Karrenbelt) [inline aea/test_tools/click_testing.py:157]: should make this `_cli`

Comment (Karrenbelt) [reply aea/test_tools/click_testing.py:157]: Should also make the `cli_options: Sequence[str]` for extra flexibility

**Review by angrybayblade — APPROVED** [review]

**Comment by Karrenbelt** [comment]
failures
- macos 3.8 - one of the known flakers
    ```
    =================================== FAILURES ===================================
    ___________________________ test_run[fake-password] ____________________________
    
    password_or_none = 'fake-password'
    
        @pytest.mark.flaky(reruns=MAX_FLAKY_RERUNS)
    >           process.expect("Start processing messages", timeout=10)
    ```
- windows 3.7 - haven't seen this before yet
    ```
    __________________ TestAeaExceptionPolicy.test_no_exceptions __________________
    
    self = <tests.test_aea.TestAeaExceptionPolicy object at 0x00000253EF08F448>
    
        def test_no_exceptions(self) -> None:
            """Test act and handle works if no exception raised."""
            t = Thread(target=self.aea.start)
            t.start()
        
            self.aea_tool.put_inbox(self.aea_tool.dummy_envelope())
            time.sleep(1)
            try:
    >           assert self.handler_called >= 2
    E           assert 0 >= 2
    E            +  where 0 = <tests.test_aea.TestAeaExceptionPolicy object at 0x00000253EF08F448>.handler_called
    ```

**Pull request merged**